test/e2e/autoscaling/cluster_size_autoscaling.go

```diff
@@ -1021,6 +1043,9 @@ func makeNodeSchedulable(c clientset.Interface, node *v1.Node) error {
 		}
 		newTaints := make([]v1.Taint, 0)
 		for _, taint := range freshNode.Spec.Taints {
+			if taint.Key == criticalAddonsOnlyTaint {
```

Comment:
I guess it doesn't matter this much, since you've added a retry loop in AfterEach for this, but I still don't like this too much. My problem is we use this function in defers to cleanup the taint we added and I would argue it would be cleaner to actually remove it regardless whether rescheduler added it's taint or not. As it is those defers feel pointless.

So I would just set some bool here and check it again before returning nil